### ISSUE TITLE
prefer-const false positive in a simple variable declaration

### ISSUE THREAD

**Issue by javasaur** [palantir/tslint#4315]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.1.6
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
let a = 5
var b = 6
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "prefer-const": true,
    }
}
```

#### Actual behavior
No error/warning is thrown
#### Expected behavior
Error/warning that variables a & b can be declared as const variables sohuld be thrown

**Comment by JoshuaKGoldberg** [comment]
Interesting. `prefer-const` won't check variables declared in a global scope like that if the file doesn't have any `import` or `export` keywords (so if the file's not a module file).

These file contents do provide errors:

```typescript
let a = 5 // prefer-const complaint
var b = 6 // prefer-const complaint

export let c = 7
```

#491, the issue for adding `prefer-const`, doesn't mention why that's the case, and neither does the preceding https://github.com/Microsoft/TypeScript/pull/5531. 

However, https://github.com/palantir/tslint/issues/3587 does mention this:

> This is working as intended. If you don't have an import or export in a file, it's not a module but is considered global. Since TSLint cannot know if some other file uses variables in global scope, prefer-const ignores variables defined at the top level of global scripts.

Marking as duplicate. Thanks for posting! TIL `prefer-const` is different for globals.
